src/form.js

```diff
@@ -187,3 +187,27 @@ export function isFieldDefault(field) {
   }
   return true;
 }
+
+/**
+ * Checks if a form field is empty.
+ * @param {!Element} field
+ * @throws {Error}
+ * @return {boolean}
+ */
+export function isFieldEmpty(field) {
+  switch (field.tagName) {
+    case 'INPUT':
+      if (field.type == 'checkbox' || field.type == 'radio') {
+        return !field.checked;
+      } else {
+        return !field.value;
+      }
+    case 'TEXTAREA':
+      return !field.value;
+    case 'SELECT':
+      // dropdown menu has at least one option always selected
+      return false;
+    default:
+      throw new Error(`${field.tagName} is not a supported field element.`);
```

Comment:
@GoTcWang Changed the conditional to `switch` on `tagName`, and also throws an error upon unsupported field element. I used this existing error message as a reference:

https://github.com/ampproject/amphtml/blob/95c5ac41e773e54630ac64e9e9ff53259a18c6c7/extensions/amp-bind/0.1/bind-expression.js#L374

Reply:
Discussed offline. This helper is only called for those elements triggering `input` event so throwing an error is the correct way to go, I think. However, we should definitely document this in the JsDoc so it won't be misused later.

Reply:
I explicitly stated the supported elements in the JsDoc, but deliberately left out the `input event` implementation detail.